Objective-C/CBLReplicator.mm

```diff
@@ -387,8 +387,11 @@ - (void) removeChangeListenerWithToken: (id<CBLListenerToken>)token {
 
 - (void) setProgressLevel: (CBLReplicatorProgressLevel)level {
     _progressLevel = level;
-    if (_repl)
-        assert(c4repl_setProgressLevel(_repl, (C4ReplicatorProgressLevel)level, nullptr));
+    if (_repl) {
+        BOOL success = c4repl_setProgressLevel(_repl, (C4ReplicatorProgressLevel)level, nullptr);
```

Comment:
Can you add an error to the last parameter as well so the error can be logged? Also, in what circumstance, this can be failed?

Reply:
only two cases when replicator is null OR level is wrong, which are already taken care or show build error in platform side.